R/utilities.R

```diff
@@ -0,0 +1,346 @@
+                pull_hifld_field(HIFLD.ID, "ADDRESS", hifld_data)), 
+            City = coalesce(
+                City, 
+                pull_hifld_field(HIFLD.ID, "CITY", hifld_data)), 
+            Zipcode = coalesce(
+                Zipcode, 
+                pull_hifld_field(HIFLD.ID, "ZIP", hifld_data)), 
+            County = coalesce(
+                County, 
+                pull_hifld_field(HIFLD.ID, "COUNTY", hifld_data)), 
+            County.FIPS = coalesce(
+                County.FIPS, 
+                pull_hifld_field(HIFLD.ID, "COUNTYFIPS", hifld_data)), 
+            Security.Level = coalesce(
+                Security.Level, 
+                pull_hifld_field(HIFLD.ID, "SECURELVL", hifld_data)), 
+            Capacity = coalesce(
+                Capacity, 
+                pull_hifld_field(HIFLD.ID, "CAPACITY", hifld_data)), 
+            Website = coalesce(
+                Website, 
+                pull_hifld_field(HIFLD.ID, "WEBSITE", hifld_data))
+        ) %>% 
+        ungroup() %>%
+        # Geocode addresses
+        tidygeocoder::geocode(
+            method = "geocodio", 
+            street = Address,
+            city = City,
+            county = County,
+            state = State,
+            postalcode = Zipcode,
+            lat = "lat_geo_",
+            long = "long_geo_"
+        ) %>%
+        mutate(
+            Latitude = coalesce(Latitude, lat_geo_),
+            Longitude = coalesce(Longitude, long_geo_)
+        ) %>%
+        select(-lat_geo_, -long_geo_) %>%
+        # Assign Facility.ID
+        mutate(Facility.ID = generate_new_fac_id() + row_number() - 1)
+}
+
+pull_hifld_field <- function(id, field, hifld_data = NULL) {
+    #' Get HIFLD data for a single field and ID  
+    #'
+    #' @param id integer HIFLD ID 
+    #' @param field character string for the HIFLD field to pull data from 
+    #' @param hifld_data data frame with HIFLD data 
+    #'
+    #' @return value for the given field and ID 
+    #'
+    #' @examples
+    #' pull_hifld_field(10002798, "CITY")
+    
+    if (is.null(hifld_data)) {
+        hifld_data <- read_hifld_data()
+    }
+    
+    if (is.na(id)) {out <- NA} 
+    else if (! is_hifld_id(id, hifld_data)) {out <- NA} 
+    else {
+        out <- hifld_data %>% 
+            filter(hifld_id == id) %>% 
+            pull(field) 
+    }
+    return (out)
+}
+
+update_fac_info <- function(new_fac_info, old_fac_info = NULL) {
+    #' Update UCLA facility data 
+    #' 
+    #' Updates the UCLA facility info sheet by combining the existing sheet with 
+    #' the new entries. 
+    #'
+    #' @param new_fac_info data frame, new facility info updates
+    #' @param old_fac_info data frame, existing facility info  
+    #' 
+    #' @return data frame, combining existing and new facilities  
+
+    if (is.null(old_fac_info)) {
+        old_fac_info <- behindbarstools::read_fac_info()
+    }
+    
+    out <- old_fac_info %>% 
+        bind_rows(new_fac_info)
```

Comment:
I'd add in `arrange(State)` here, to make it a little easier for manual searches